src/resource.jl

```diff
@@ -344,13 +366,14 @@ function serve_file(
 
     # Open the path, write it out directly to the HTTP stream in chunks
     open(path) do io
+        seek(io, startbyte)
         t = 0
-        while !eof(io)
+        while t < content_length
             n = readbytes!(io, buffer)
```

Comment:
```suggestion
            r = min(length(buffer), content_length - t)
            n = readbytes!(io, buffer, r)
```
This avoids reading the entire buffer when less data is needed.

Reply:
Yea. Does it work though? I just saw https://github.com/JuliaLang/julia/issues/36300. I will just leave it as a comment for now, can be optimized later.